contestreminder/users/urls.py

```diff
@@ -1,6 +1,9 @@
 from django.urls import path
-from .views import SignUpView
+from django.contrib.auth.decorators import login_required
+from .views import SignUpView, ProfileView, DeleteUserView
 
 urlpatterns = [
     path('signup/', SignUpView.as_view(), name='signup'),
-]
\ No newline at end of file
+    path('profile/', login_required(ProfileView.as_view()), name='profile'),
+    path('delete/', login_required(DeleteUserView.as_view()), name='delete'),
```

Comment:
1) pass login_url='/users/login' as an argument.

If you will try to access profile page then you will see something like this 
"http://127.0.0.1:8000/accounts/login/?next=/users/profile/"

By default, login_url is /accounts/login so we need to change that.

Reply:
Would it be best to have `login_url` passed for both `login_required` function calls (both the `/profile` and `delete/` paths)?

Or should it be just for `/profile`?

Reply:
Yes It we need to pass login_url for both.

If someone try to access any of above url(by editing path in browser's search bar) without logging in then it will throw an error.

Try accessing url without logging in, you will get the better idea.